Merging dgi code analyzer with minerva java analyzer and improvements done for sdg generation

fabio-franco opened the pull request:
As part of this task, there are 5 new messages created:

```
CallGraphFileIsNotJAR = The file {0} is not a JAR and it will be skipped.
CallGraphAddExtraLibToScope = Adding dependency library to analysis scope: {0}
CallGraphBuildMethodLevel = Building the call graph with method level information ...
CallGraphPopulatingMethodLevel = Populating call graph with method level information ...
CallGraphCalculatingWeightMethodLevel = Calculating the weight for the connections in the call graph with method level information ...
```

fabio-franco committed:
First merge between dgi-code-analyzer and minerva binary analyzer

fabio-franco committed:
Updating the license headers

fabio-franco committed:
Adding extra libs feature for the binary analyzer

fabio-franco committed:
Improving messages and cleaning up logs

fabio-franco committed:
Cleaning up small details

fabio-franco committed:
Changing log level for extra lib file is not a jar message

fabio-franco committed:
Merge pull request #1 from fabio-franco/merging_dgi_code_analyzer_with_minerva_java_analyzer

Merge between dgi-code-analyzer and minerva binary analyzer

fabio-franco committed:
Adding missing log for not jar file

fabio-franco committed:
Changing entry points and making the analysis more robust (using phantom to the class hierarchy)

fabio-franco committed:
Changing the code to use makeWithRoot and removing AnalysisUtils

fabio-franco committed:
Changes to capture parcial time and to test phantom with recent fixes proposals

fabio-franco committed:
Fixes and improvements for sdg graph generation

fabio-franco commented on src/main/java/com/ibm/minerva/analyzer/Analyzer.java:
```diff
@@ -95,9 +111,42 @@ public Analyzer setCallGraphBuilder(CallGraphBuilderType type) throws IOExceptio
 
     public void run() throws IOException {
         try {
+        	long start = System.currentTimeMillis();
```
need to be removed

lentheivendra replied:
As discussed, can remove all temporary time captures etc

fabio-franco commented on src/main/java/com/ibm/minerva/analyzer/Analyzer.java:
```diff
@@ -118,7 +167,16 @@ public void run() throws IOException {
+            
+            long end = System.currentTimeMillis();
+            
+            logger.info("Time (in milliseconds) to process all classes: "+(end-start));
```
need to be removed

lentheivendra commented on src/main/resources/com/ibm/minerva/analyzer/messages.properties:
```diff
@@ -51,3 +51,9 @@ CallGraphTempFileWriteWarning = Class {0} will be processed in memory because it
 CallGraphBuildError = An unexpected error occurred while attempting to build the call graph. Message: {0}
 CallGraphWriteError = An unexpected error occurred while attempting to write the call graph to file {0}. Message: {1}
+
+CallGraphFileIsNotJAR = The file {0} is not a JAR and it will be skipped.
```
Just need to drop the "it"

lentheivendra reviewed (approved):
LGTM

fabio-franco committed:
Addressing pr comments

fabio-franco committed:
Small fix to avoid nullPointerException in jpet app

During a clean up I removed these verifications, but they are important to avoid a nullpointerexception in jpet

fabio-franco commented:
During a clean up I removed the verifications inside the while loop in SDGGraph2JSON.java, but they are important to avoid a null pointer exception in jpet, so I am returning back those extra verifications.

fabio-franco committed:
Fix to remove interfaces with inner classes

fabio-franco committed:
Merge branch 'main' of github.com:fabio-franco/codenet-minerva-java-analyzer into merging_dgi_code_analyzer_with_minerva_java_analyzer

fabio-franco committed:
fixing bug on data dependency

fabio-franco committed:
Merge pull request #2 from fabio-franco/fix_for_interface_with_inner_class

Fix to remove interfaces with inner classes

fabio-franco committed:
Merge branch 'main' of github.com:fabio-franco/codenet-minerva-java-analyzer into merging_dgi_code_analyzer_with_minerva_java_analyzer
